syntax = "proto3";
package wilson;

import "properties.proto";

//Text positioning
enum TextPosition {
    CENTER = 0;      //Center of screen
    UPPER_RIGHT = 1; //Upper right corner
    TOP = 2;         //Middle of top side
    UPPER_LEFT = 3;  //Upper left corner
    LEFT = 4;        //Middle of the left side
    LOWER_LEFT = 5;  //Lower left corner
    BOTTOM = 6;      //Middle of the bottom side
    LOWER_RIGHT = 7; //Lower right corner
    RIGHT = 8;       //Middle of right side
}

//Overlay text drawn on the scene surface
message Overlay {
    //Text to be drawn on the scene
    string text = 1;

    //Position of the text
    TextPosition position = 2;
    //Font style
    ScalarProperty fontSize = 3;
    bool bold = 4;
    bool italic = 5;
}
